packages/devextreme/js/__internal/grids/new/grid_core/data_controller/utils.ts

```diff
@@ -31,6 +33,110 @@ export function normalizeDataSource(
   return new DataSource(normalizeDataSourceOptions(dataSourceLike, undefined));
 }
 
+export function isLocalStore(store: Store): boolean {
+  return store instanceof ArrayStore;
+}
+
+export function isCustomStore(store: Store): boolean {
+  return store instanceof CustomStore;
+}
+
+export function normalizeRemoteOptions(
+  remoteOperations: boolean | object | string,
+  localStore: boolean,
+  customStore: boolean,
+): OperationOptions {
+  if (remoteOperations instanceof String && remoteOperations !== 'auto') {
+    throw new Error('Remote operations do not support any string values except \'auto\'');
+  }
```

Comment:
Let's move this check to TS and specify more precise type of `remoteOperations` instead of `object | string`